pkg/util/rand/rand.go

```diff
@@ -54,3 +54,11 @@ func Seed(seed int64) {
 
 	rng.rand = rand.New(rand.NewSource(seed))
 }
+
+// Perm returns, as a slice of n ints, a pseudo-random permutation of the integers [0,n)
+// from the default Source.
+func Perm(n int) []int {
+	rng.Lock()
```

Comment:
Who seeds this RNG?  I don't see a call to Seed() in here - isn't this the same problem, just one extra layer?